### PR TITLE
Delete a range of values from temp blob list

### PR DESCRIPTION
**Pull request by adrogin**
Currently, there is no way to efficiently delete a range from a Temp Blob List. If I need to clear a list, I can call RemoveAt in a loop, but this method copies a bulk of elements on each iteration.
This pull request introduces the RemoveRange operation, identical to the List data type, to remove a range in a single operation.

### CODE REVIEW

File: Modules/System/BLOB Storage/src/TempBlobList.Codeunit.al [hunk 1]
```diff
@@ -67,6 +67,18 @@ codeunit 4102 "Temp Blob List"
         exit(TempBlobListImpl.RemoveAt(Index));
     end;
 
+    /// <summary>
+    /// Remove a number of elements starting from a specified location from a non-empty list.
+    /// </summary>
+    /// <error>The index is larger than the number of elements in the list or less than one.</error>
+    /// <param name="Index">The index of the first element to be deleted.</param>
+    /// <param name="ElementCound">The number of elements to be deleted.</param>
```
Comment (mazhelez) [inline Modules/System/BLOB Storage/src/TempBlobList.Codeunit.al:75]: Typo

```suggestion
    /// <param name="ElementCount">The number of elements to be deleted.</param>
```

File: Modules/System/BLOB Storage/src/TempBlobListImpl.Codeunit.al [hunk 2]
```diff
@@ -44,13 +44,20 @@ codeunit 4109 "Temp Blob List Impl."
...
     var
         TempBlobList: Codeunit "Temp Blob List";
     begin
         if not TempBlobRec.Get(Index) then
             Error(ObjectDoesNotExistErr, Index);
-        if TempBlobRec.Get(Index + 1) then
-            GetRange(Index + 1, Count() - Index, TempBlobList);
+
+        if TempBlobRec.Get(Index + ElementCount) then
+            GetRange(Index + ElementCount, Count() - Index - ElementCount, TempBlobList);
```
Comment (mazhelez) [inline Modules/System/BLOB Storage/src/TempBlobListImpl.Codeunit.al:59]: This PR about is parametrizing the `1`. 

However, `Count() - Index` became `Count() - Index - ElementCount`. Why is that? Was this an issue before?

File: Modules/System/BLOB Storage/src/TempBlobListImpl.Codeunit.al [hunk 3]
```diff
@@ -8,9 +8,9 @@ codeunit 4109 "Temp Blob List Impl."
     Access = Internal;
 
     var
-    #pragma warning disable AA0073
+#pragma warning disable AA0073
```
Comment (mazhelez) [inline Modules/System/BLOB Storage/src/TempBlobListImpl.Codeunit.al:11]: Formatting is weird here

File: Modules/System/BLOB Storage/src/TempBlobList.Codeunit.al (hunk 1 above)
Comment (mazhelez) [reply Modules/System/BLOB Storage/src/TempBlobList.Codeunit.al:75]: Also, what if negative value is passed? Is this supported?

**Review by mazhelez — COMMENTED** [review]
Minor comments to address. But overall: great contribution. 👍

File: Modules/System/BLOB Storage/src/TempBlobList.Codeunit.al (hunk 1 above)
Comment (adrogin) [reply Modules/System/BLOB Storage/src/TempBlobList.Codeunit.al]: Thanks for the comment. Added an explicit error and test cases for a negative count.

File: Modules/System/BLOB Storage/src/TempBlobListImpl.Codeunit.al (hunk 2 above)
Comment (adrogin) [reply Modules/System/BLOB Storage/src/TempBlobListImpl.Codeunit.al]: That's a bug, indeed. Existing tests did not cover this verification. Fixed and added tests.

File: Modules/System/BLOB Storage/src/TempBlobListImpl.Codeunit.al (hunk 3 above)
Comment (adrogin) [reply Modules/System/BLOB Storage/src/TempBlobListImpl.Codeunit.al:11]: That's AL Language extension autoformatting. And as far as I can see, the pragma instruction is always aligned on the left without indentation.

Comment (pri-kise) [reply Modules/System/BLOB Storage/src/TempBlobListImpl.Codeunit.al:11]: That formatting was reported on the AL Language Repository, but it wasn't solved yet.
See Issue https://github.com/microsoft/AL/issues/6530

**Review by JesperSchulz — APPROVED** [review]

**Comment by JesperSchulz** [comment]
Thanks for reporting this. We agree, and we’ll publish a fix asap, either in an update for the current version or in the next major release. Please do not reply to this, as we do not monitor closed issues. If you have follow-up questions or requests, please create a new issue where you reference this one.

Build ID: 49407.

**Comment by JesperSchulz** [comment]
Availability update: We will publish a fix for this issue in the next update for release 21.

Build ID to track: 49535.

**Pull request merged**